pkg/master/controller.go

```diff
@@ -212,26 +227,10 @@ func (c *Controller) SetEndpoints(serviceName string, ip net.IP, port int) error
 		glog.Warningf("Resetting endpoints for master service %q to %v", serviceName, e)
 		return c.EndpointRegistry.UpdateEndpoints(ctx, e)
 	} else if !ipCorrect {
-		// We *always* add our own IP address; if there are too many IP
-		// addresses, we remove the ones lexicographically after our
-		// own IP address.  Given the requirements stated at the top of
-		// this function, this should cause the list of IP addresses to
-		// become eventually correct.
 		e.Subsets[0].Addresses = append(e.Subsets[0].Addresses, api.EndpointAddress{IP: ip.String()})
 		e.Subsets = endpoints.RepackSubsets(e.Subsets)
-		if addrs := &e.Subsets[0].Addresses; len(*addrs) > c.MasterCount {
```

Comment:
Why are you removing this section? This will break the service, because nothing will generate endpoints for it? nodePort services are still implemented by kube-proxy getting these endpoints and routing traffic to them!